### DISCUSSION TITLE
scrollBehaviour: Target Element

### DISCUSSION THREAD

**Discussion by bennyschudel** [vuejs/router#2249] [Ideas]
Hello

I noticed that `scrollBehaviour` does only work with the document but not with a container. However I have applications that need the restoration of the scroll-position within another element.

I had a look at the code and for me it seems to be possible without much effort.

Would you think I would be feasible to add a target property?

What are your thoughts on that?

Any reasons this has not been implemented in the first-hand?

Best
Benny

**Comment by posva** [comment]
Duplicate of https://github.com/vuejs/vue-router/issues/1187

**Comment by posva** [comment]
My plan is to adapt the API in the future to be more flexible. You can use https://github.com/vuejs/vue-router/issues/1187#issuecomment-1583272925 in the meantime
